src/video/SDL_video.c

```diff
@@ -630,10 +630,21 @@ SDL_DisplayID SDL_AddVideoDisplay(const SDL_VideoDisplay *display, SDL_bool send
     SDL_VideoDisplay *displays, *new_display;
     SDL_DisplayID id = 0;
 
-    displays = (SDL_VideoDisplay *)SDL_realloc(_this->displays, (_this->num_displays + 1) * sizeof(*displays));
+    displays = (SDL_VideoDisplay *)SDL_malloc((_this->num_displays + 1) * sizeof(*displays));
     if (displays) {
         int i;
+
+        /* The display list may contain self-referential pointers to the desktop mode. */
+        for (i = 0; i < _this->num_displays; ++i) {
```

Comment:
The last element of the array is not set.
Or does SDL_malloc/SLD_calloc zero initialize memory?

I can't immediately see this on the wiki.

Reply:
Never mind; it is set in lower code.
Sorry about the noise.